apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClientPartial.cs

```diff
@@ -23,6 +23,32 @@
 
 namespace Google.Cloud.Bigtable.V2
 {
+    public partial class BigtableServiceApiSettings
+    {
+        partial void OnConstruction()
```

Comment:
I thought we were changing MutateRowsSettings to retry by default, now that it's seen as idempotent? Maybe I've missed something. I'm most fine with this as a temporary measure anyway - but it does mean that if a caller has modified the default settings (e.g. to use different credentials) then this would fail, right?

Reply:
Yes, that's the idea. Perhaps I misunderstood this part of it though. I assumed we wanted to retry by default but still let users control the retry settings, which I would assume would mean we need to update the default call settings for MutateRows. Is there a better way to do this? Or do we not want to let users customize the retry settings?

As far as specifying other settings, I realize I was only considering per-call settings, which would merge correctly with these default retry settings, and therefore now replace them. But if they change the settings when creating the client, yes this would fail.

Reply:
Also, how does this work in general. If you want to use different credentials with MutateRow, which has auto-generated settings that retry by default now, wouldn't that also prevent retry?

Reply:
If the GAPIC config is changing to make MutateRows an idempotent call, the default settings should automatically change to allow retries. On the other hand, it could well be that I've missed something, particularly with it being a streaming method.

If we're only trying to change the timing, maybe we should not check anything else, and just use `originalMutateRowsSettings.WithTiming(...)`?

Reply:
Sorry, hadn't seen the second comment. I'd expect *most* users to only specify a CallSettings on a per-call basis for modifications, but it's possible that some will modify the default settings instead. It's possible that what's currently in the master branch won't work with this properly and that I failed to review properly before.

We should also consider what we want to happen if the user specifies a custom CallSettings because they want to change the timing, but they *do* still want retry. Oh, it's confusing :(

Reply:
Oh, I see. Sorry, I misunderstood your "retry by default" comment originally to mean functionally, not in the auto-generated stuff. So the config is updated: https://github.com/googleapis/googleapis/blob/master/google/bigtable/v2/bigtable_gapic.yaml#L104, but the generator did not generate retry settings: https://github.com/evildour/google-cloud-dotnet/blob/master/apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClient.cs#L183. I'm guessing this is because it is a streaming API so that goes through a different path. @chrisdunelm can you confirm?

Yes, using `WithCallTiming` seems much cleaner. I'll switch over to that if this `OnConstruction` is still needed.

As for my "how does this work in general" comment, I'm not even talking about what's there for MutateRow specifically. Take `FirestoreClient.GetDocument` for example: that has default idempotent retry settings. If a caller has modified the default settings to use different credentials for that method, retry for that would fail too right?

Reply:
It depends on how they modify the default settings. IIRC, we throw an exception if you try to create an ApiCall using base settings that have *no* timing - but if the client decided to create a timing that didn't include any retry then yes, that wouldn't retry.

More likely, they'd use `existingCallSettings.WithCredentials(...)` to create the new one.

Reply:
Ok, so it sounds like the same thing should happen here (unless the validation for call timing doesn't happen for the streaming calls). So it sounds like it's safe to say your original concern is not an issue, but correct me if I'm wrong.

The only question left now is why we're not auto-generating these retry settings despite the GAPIC config specifying them. I'll update this code to with WithCallTiming as you suggest while we wait for clarification on that.

Reply:
Right - the difference is that we can't easily detect "the user has changed the call timing and we should obey that" vs "we should replace the call timing with a retry". But if we get to the bottom of the autogeneration diff, it may not matter.